R/pr.R

```diff
@@ -74,22 +80,37 @@ pr_init <- function(branch) {
 #'   pull request. Default of `NULL` tries to identify the source repo and uses
 #'   the owner of the `upstream` remote, if present, or the owner of `origin`
 #'   otherwise.
+#' @inheritParams git2r_credentials
 #'
 #' @examples
 #' \dontrun{
 #' ## scenario: current project is a local copy of fork of a repo owned by
 #' ## 'tidyverse', not you
 #' pr_fetch(123, owner = "tidyverse")
 #' }
-pr_fetch <- function(number, owner = NULL) {
+pr_fetch <- function(number,
+                     owner = NULL,
+                     credentials = NULL,
+                     auth_token = github_token()) {
+  check_uses_github()
   check_uncommitted_changes()
+  check_github_token(auth_token)
+
+  protocol <- github_remote_protocol()
+  credentials <- credentials %||% git2r_credentials(protocol, auth_token)
 
-  ui_done("Retrieving data for PR #{number}")
-  pr <- gh::gh("GET /repos/:owner/:repo/pulls/:number",
-    owner = owner %||% github_source() %||% github_owner(),
-    repo = github_repo(),
-    number = number
+  owner <- owner %||% github_owner_upstream() %||% github_owner()
+  repo <- github_repo()
+  pr_string <- glue("{owner}/{repo}/#{number}")
+  ui_done("Retrieving data for PR {ui_value(pr_string)}")
```

Comment:
Having two messages for this one action seems a bit heavy to me.

Reply:
It looks really nice though 🙂. I can combine them into 1, issued after the GET, so the interesting content is there.